### DISCUSSION TITLE
private final SecurityUtil securityUtil 를 Controller 에서 두는게 맞는지 Service에 두는게 맞는지??

### DISCUSSION THREAD

**Discussion by Aleexender** [SCBJ-7/SCBJ-BE#93] [Ideas]
```
@Component
public class SecurityUtil {
    public Long getCurrentMemberId() {
        final Authentication authentication = SecurityContextHolder.getContext()
            .getAuthentication();
        if (authentication == null || !authentication.isAuthenticated()) {
            throw new RuntimeException("Security Context에 인증 정보가 없습니다.");
        }

        return Long.parseLong(authentication.getName());
    }
}
```

SecurityUtils를 사용할때 Controller 단에다가 작성하는게 맞을까요? 아니면 Service단에 두는게 맞을까요??

저는 OOP적으로 보았을때 Controller 단에서 작성하는게 좋을꺼같은 느낌이에요!
해당 이유는 Controller에서 작성할경우 비지니스 로직에서의 수정 혹은 기능 추가가 생겼을때 Controller단에서는 Service 에서 id만 보내주면 끝나기때문에 OOP 적으로 보면 Controller단에서 보내주는게 맞다고 생각하는데 반대 입장에서는 해당 SecurityUitils도 하나의 비지니스 로직이기때문에 Service 에 넣어야한다로 보이는데 다들 어떻게 생각하시나요??

![image](https://github.com/SCBJ-7/SCBJ-BE/assets/111270670/71622a8d-d38e-4677-a0e7-549b58f3ca8f)

**Comment by YurimYang** [comment]
흠.. 어려운 주제인 것 같아요.
이부분을 생각하기 이전까지는 저도 항상 Controller에 작성해오긴 했어요. 하지만 개인적으로 SecurityUtil의 get메소드가 서비스 로직이라서 Service 로직에서 작동하는게 좋을 것 같다고 생각합니다.

**Comment by Kwonminwoo** [comment]
요청 유저 검증 과정이니까 컨트롤러에서 처리 되는게 맞다고 생각합니다
